pkg/internal/checkup/checkup.go

```diff
@@ -283,6 +287,31 @@ func newTrafficGeneratorPod(checkupConfig config.Config, secondaryNetworkRequest
 	)
 }
 
+func (c *Checkup) waitForPodDeletion(ctx context.Context) error {
+	if c.trafficGeneratorPod == nil {
+		return fmt.Errorf("failed to delete traffic generator Pod, object doesn't exist")
+	}
+
+	podFullName := ObjectFullName(c.trafficGeneratorPod.Namespace, c.trafficGeneratorPod.Name)
+	log.Printf("Waiting for Pod %q to be deleted..", podFullName)
+
+	conditionFn := func(ctx context.Context) (bool, error) {
+		var err error
+		_, err = c.client.GetPod(ctx, c.trafficGeneratorPod.Namespace, c.trafficGeneratorPod.Name)
+		if k8serrors.IsNotFound(err) {
+			return true, nil
+		}
+		return false, err
+	}
+	const interval = time.Second * 5
```

Comment:
nit: Could you please put the literal in front of the constant?
```suggestion
	const interval = 5 * time.Second
```

This way it reads as "5 seconds".

Reply:
DONE